src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -900,6 +901,28 @@ public void testSubstituteInnerClass() {
 		assertEquals("x.Result$Inner",fr.getDeclaringType().getQualifiedName());
 	}
 
+	@Test
+	public void testSubstitutionInsertAllFields() {
+		// contract: insertAllFields method of the Substitution class inserts all the fields from a given template by substituting all the template parameters by their values.
```

Comment:
This contract does not adequately describe the test. What _actually_ happens in this test is that a single field, `String testString = "goodName";` is inserted into the created class. There is no substitution going on, there is only a single as-is field insertion.

A more apt description would be something like "Substitution.insertAllFields inserts the only field from a single-field template into the target class"

Additionally, the test actually only verifies that the field has the correct name, so it doesn't verify the field as a whole.

Reply:
Yes, I have changed the contract, I thoughtt I was supposed to write what the method is doing in the contract, than what is test is supposed to do.

> test actually only verifies that the field has the correct name, so it doesn't verify the field as a whole.

Yes, I have added another assertion for checking the assignment of the field.

Reply:
The contract should state any relevant preconditions for the test, method/class/etc is being tested, and what the expected out come is. How one gets from the preconditions to the expected outcome is an implementation detail a test in general is not concerned with :)

Sometimes it may be necessary to test an implementation detail, especially when testing corner cases or reproducing bugs. Then it may be relevant to explain such things in the contract. So, as with everything, this is not a rule set in stone, but rather a general guideline.

Assume that pretty much all of my feedback is either a general guideline, or just my opinion, or both.